units/host_termination.go

```diff
@@ -227,36 +227,33 @@ func (j *hostTerminationJob) Run(ctx context.Context) {
 		return
 	}
 
-	if j.host.Status != evergreen.HostProvisionFailed {
-		// only run teardown if provisioning was successful
-		if err := runHostTeardown(ctx, j.host, cloudHost); err != nil {
-			grip.Error(message.WrapError(err, message.Fields{
-				"job_type": j.Type().Name,
-				"message":  "Error running teardown script",
-				"host":     j.host.Id,
-			}))
+	if err := j.runHostTeardown(ctx, cloudHost); err != nil {
+		grip.Error(message.WrapError(err, message.Fields{
+			"job_type": j.Type().Name,
+			"message":  "Error running teardown script",
+			"host":     j.host.Id,
+		}))
 
-			subj := fmt.Sprintf("%v Error running teardown for host %v",
-				teardownFailurePreface, j.host.Id)
-
-			mailer, serr := j.env.GetSender(evergreen.SenderEmail)
-			if serr != nil {
-				grip.Alert(message.Fields{
-					"message":    "problem getting sender",
-					"operation":  "host termination issue",
-					"sender_err": serr,
-					"error":      err,
-					"host":       j.host.Id,
-					"subject":    subj,
-				})
-			} else {
-				mailer.Send(message.NewEmailMessage(level.Error, message.Email{
-					From:       settings.Notify.SMTP.From,
-					Recipients: settings.Notify.SMTP.AdminEmail,
-					Subject:    subj,
-					Body:       err.Error(),
-				}))
-			}
+		subj := fmt.Sprintf("%v Error running teardown for host %v",
+			teardownFailurePreface, j.host.Id)
+
+		mailer, serr := j.env.GetSender(evergreen.SenderEmail)
+		if serr != nil {
+			grip.Alert(message.Fields{
+				"message":    "problem getting sender",
+				"operation":  "host termination issue",
+				"sender_err": serr,
+				"error":      err,
+				"host":       j.host.Id,
+				"subject":    subj,
+			})
+		} else {
+			mailer.Send(message.NewEmailMessage(level.Error, message.Email{
+				From:       settings.Notify.SMTP.From,
+				Recipients: settings.Notify.SMTP.AdminEmail,
```

Comment:
I'd recommend checking if settings.Notify.SMTP.AdminEmail is empty

Reply:
Thanks, doing this.